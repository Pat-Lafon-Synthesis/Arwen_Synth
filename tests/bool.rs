use std::{fs::File, io::Read};

use arwen_synth::{parser_interface::parse, synthesis};

mod libraries;
use libraries::bool_library;

macro_rules! make_test {
    ($test_name:tt, $($libs:tt)*) => {
        #[test]
        fn $test_name() {
            let mut file =
                File::open(format!("tests/benchmarks/{}.mls", stringify!($test_name))).unwrap();
            let mut buffer = String::new();
            file.read_to_string(&mut buffer).unwrap();
            let synth_problem = parse(buffer);

            let prog = synthesis(
                synth_problem.sig.into(),
                $($libs)*,
                synth_problem.tests.tests.into(),
                3,
            );
            insta::assert_display_snapshot!(stringify!($test_name), prog.unwrap());
        }
    };
}

make_test!(bool_always_false, &bool_library());
make_test!(bool_always_true, &bool_library());
make_test!(bool_band, &bool_library());
make_test!(bool_bor, &bool_library());
make_test!(bool_impl, &bool_library());
make_test!(bool_neg, &bool_library());
make_test!(bool_xor, &bool_library());
