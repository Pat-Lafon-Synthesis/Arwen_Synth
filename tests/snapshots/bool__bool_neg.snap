---
source: tests/bool.rs
expression: prog.unwrap()
---
not(arg0)
